pkg/controller/registry/registry_client.go

```diff
@@ -97,25 +100,56 @@ func (rc *Client) FindBundleThatProvides(ctx context.Context, group, version, ki
 	return bundle, nil
 }
 
-// FilterChannelEntries filters out a channel entries that provide the requested
+// FilterChannelEntries filters out channel entries that provide the requested
 // API and come from the same package with original operator and returns the
-// first entry on the list
-func (rc *Client) filterChannelEntries(it *ChannelEntryIterator, excludedPkgName string) *opregistry.ChannelEntry {
+// first entry on the list from the default channel of that package
+func (rc *Client) filterChannelEntries(ctx context.Context, it *ChannelEntryIterator, excludedPkgName string) (*opregistry.ChannelEntry, error) {
+	defChannels := make(map[string]string, 0)
+
 	var entries []*opregistry.ChannelEntry
 	for e := it.Next(); e != nil; e = it.Next() {
 		if e.PackageName != excludedPkgName {
-			entry := &opregistry.ChannelEntry{
-				PackageName: e.PackageName,
-				ChannelName: e.ChannelName,
-				BundleName:  e.BundleName,
-				Replaces:    e.Replaces,
+			// keep track of the default channel for each package
+			if _, ok := defChannels[e.PackageName]; !ok {
+				defChannel, err := rc.getDefaultPackageChannel(ctx, e.PackageName)
+				if err != nil {
+					return nil, err
+				}
+				defChannels[e.PackageName] = defChannel
+			}
+
+			// only add entry to the list if the entry is in the default channel
+			if e.ChannelName == defChannels[e.PackageName] {
+				entry := &opregistry.ChannelEntry{
+					PackageName: e.PackageName,
+					ChannelName: e.ChannelName,
+					BundleName:  e.BundleName,
+					Replaces:    e.Replaces,
+				}
+				entries = append(entries, entry)
 			}
-			entries = append(entries, entry)
 		}
 	}
 
 	if entries != nil {
```

Comment:
Nit: I will sleep better if this becomes a length-based condition, since entries can always be a non-nil but empty slice.